src/storage/aggregator.cc

```diff
@@ -143,6 +159,106 @@ bool Aggregator::Update(const std::string& key, const std::string& row, const ui
     return true;
 }
 
+bool Aggregator::Init() {
+    if (GetStat() != AggrStat::kUnInit) {
+        PDLOG(INFO, "aggregator is normal or recovering");
+        return true;
+    }
+
+    status_.store(AggrStat::kRecovering, std::memory_order_relaxed);
+    if (!base_replicator_) {
+        status_.store(AggrStat::kUnInit, std::memory_order_relaxed);
+        return false;
+    }
+    auto log_parts = base_replicator_->GetLogPart();
+    if (aggr_table_->GetRecordCnt() == 0 && log_parts->IsEmpty()) {
+        status_.store(AggrStat::kInited, std::memory_order_relaxed);
+        return true;
+    }
+    auto table_it = aggr_table_->NewTraverseIterator(0);
+    // TODO: need abstract TraverseIterator?
+    auto it = dynamic_cast<MemTableTraverseIterator*>(table_it);
```

Comment:
Is it better to add abstract TraverseIterator for use `NextPK()` to iterate between keys?